File: C++/stack/stack.cpp
#include<iostream>
#include<vector>
using namespace std;
#define ll long long
class stack
{
    int* arr;
    int topp;

    public:
    stack(){
        arr=new int[10000];
        topp=-1;
    }

    void push(int val)
    {
        if(topp==9999)
        {
            cout<<"Stack overflow\n";
            return;
        }
        topp++;
        arr[topp]=val;
    }

    void pop()
    {
        if (topp<0)
        {
            cout<<"stack is empty\n";
            return;
        }
        
        topp--;
    }

    int top()
    {
        if(topp==-1)
        {
            cout<<"Stack empty\n";
            return -1;
        }
        return arr[topp];
    }

    bool empty()
    {
        if(topp==-1)
        {
            return true;
        }
        return false;
    }
};
main()
{
   stack st;
   st.push(1);
   st.push(2);
   st.push(3);
   st.push(4);
   st.push(5);
   st.push(6);
   cout<<st.top()<<endl;
}

File: Java_Programs_for_beginners/Stack.java
import java.util.*;
public class Stack {
    static int arr[] = new int[5];
    static int updated[];
    static int top = 0;
    static Scanner sc = new Scanner(System.in);
    public static void push() {
        if (top < arr.length) {
            System.out.println("Enter number");
            int a = sc.nextInt();
            arr[top] = a;
            top++;
        } else {
            System.out.println("Array is full");
        }
    }
    public static void pop() {
        if (top > 0) {
            top--;
        } else {
            System.out.println("Empty Stack");
        }
    }
    public static void display() {
        for (int i = 0; i < top; i++) {
            System.out.print(arr[i] + " ");
        }
        System.out.println();
    }
    public static void main(String[] args) {
        System.out.println("Enter 1 to push an element");
        System.out.println("Enter 2 to pop an element");
        System.out.println("Enter 3 to display elements");
        System.out.println("Enter 4 to exit");
        int x = sc.nextInt();
        switch (x) {
            case 1:
                push();
                main(args);
                break;
            case 2:
                pop();
                main(args);
                break;
            case 3:
                display();
                main(args);
                break;
            case 4:
                break;
            default:
                System.out.println("Enter a valid number");
                main(args);
                break;
        }
        sc.close();
    }
}
